src/vs/workbench/contrib/notebook/browser/controller/coreActions.ts

```diff
@@ -156,8 +156,21 @@ export abstract class NotebookAction extends Action2 {
 	}
 
 	async run(accessor: ServicesAccessor, context?: any, ...additionalArgs: any[]): Promise<void> {
-		const isFromUI = !!context;
-		const from = isFromUI ? (this.isNotebookActionContext(context) ? 'notebookToolbar' : 'editorToolbar') : undefined;
+		let from;
+		if (context.source) {
+			from = context.source;
+		} else if (URI.isUri(context)) {
+			from = 'cellEditorContextMenu';
+		} else {
+			const contextJson = JSON.stringify(context);
```

Comment:
This is too costly. We can do a simple `context && 'from' in context  && context.from === 'cellContainer'`.